frontend/public/components/namespace.jsx

```diff
@@ -124,149 +128,376 @@ const fetchNamespaceMetrics = () => {
+      sortFunc: 'namespaceCPU',
+      transforms: [sortable],
+      props: { className: projectAndNamespaceColumnInfo.cpu.classes },
+    },
+    {
+      title: projectAndNamespaceColumnInfo.created.title,
+      id: projectAndNamespaceColumnInfo.created.id,
       sortField: 'metadata.creationTimestamp',
       transforms: [sortable],
-      props: { className: namespacesColumnClasses[3] },
+      props: { className: projectAndNamespaceColumnInfo.created.classes },
+    },
+    {
+      title: projectAndNamespaceColumnInfo.description.title,
+      id: projectAndNamespaceColumnInfo.description.id,
+      sortField: "metadata.annotations.['openshift.io/description']",
+      transforms: [sortable],
+      props: { className: projectAndNamespaceColumnInfo.description.classes },
+      additional: true,
+    },
+    {
+      title: projectAndNamespaceColumnInfo.labels.title,
+      id: projectAndNamespaceColumnInfo.labels.id,
+      sortField: 'metadata.labels',
+      transforms: [sortable],
+      props: { className: projectAndNamespaceColumnInfo.labels.classes },
+      additional: true,
     },
-    { title: '', props: { className: namespacesColumnClasses[4] } },
+    { title: '', props: { className: Kebab.columnClass } },
   ];
 };
 NamespacesTableHeader.displayName = 'NamespacesTableHeader';
 
-const NamespacesTableRow = ({ obj: ns, index, key, style }) => {
-  return (
-    <TableRow id={ns.metadata.uid} index={index} trKey={key} style={style}>
-      <TableData className={namespacesColumnClasses[0]}>
-        <ResourceLink kind="Namespace" name={ns.metadata.name} title={ns.metadata.uid} />
-      </TableData>
-      <TableData className={classNames(namespacesColumnClasses[1], 'co-break-word')}>
-        <Status status={ns.status.phase} />
-      </TableData>
-      <TableData className={namespacesColumnClasses[2]}>
-        <LabelList kind="Namespace" labels={ns.metadata.labels} />
-      </TableData>
-      <TableData className={namespacesColumnClasses[3]}>
-        <Timestamp timestamp={ns.metadata.creationTimestamp} />
-      </TableData>
-      <TableData className={namespacesColumnClasses[4]}>
-        <ResourceKebab actions={nsMenuActions} kind="Namespace" resource={ns} />
-      </TableData>
-    </TableRow>
+const NamespacesColumnManagementID = referenceForModel(NamespaceModel);
+
+const getNamespacesSelectedColumns = () => {
+  return new Set(
+    NamespacesTableHeader().reduce((acc, column) => {
+      if (column.id && !column.additional) {
+        acc.push(column.id);
+      }
+      return acc;
+    }, []),
   );
 };
 
-export const NamespacesList = (props) => (
-  <Table
-    {...props}
-    aria-label="Namespaces"
-    Header={NamespacesTableHeader}
-    Row={NamespacesTableRow}
-    virtualize
-  />
+const namespacesRowStateToProps = ({ UI }) => ({
+  metrics: UI.getIn(['metrics', 'namespace']),
+  selectedColumns: UI.getIn(['columnManagement']),
+});
+
+const NamespacesTableRow = connect(namespacesRowStateToProps)(
```

Comment:
If the table columns are identical (which I think is fine), it would be nice to have a single `NamespaceTableRow` that handles both.

Reply:
@spadgett In `ProjectTableRow`, there are 2 props named `showMetrics` and `showActions`, while they don't exist in `NamespacesTableRow`. Do you think we need to keep it or not if we only use `NamespacesTableRow`?

Reply:
Can we just add the two props to `NamespacesTableRow`?

Reply:
As long as we have correct defaults for namespaces?

Reply:
It's OK to add these 2 props to `NamespacesTableRow`, but there are also `customData` in `ProjectTableRow`, do we also need to add that to `NamespacesTableRow`?

Reply:
@DaoDaoNoCode Let's do that as a follow on since it's feature freeze today.

Reply:
We can keep them separate for now.

Reply:
@spadgett adding bug to track this: https://bugzilla.redhat.com/show_bug.cgi?id=1862611